### ISSUE TITLE
Removing index unexpectedly drops foreign key

### ISSUE THREAD

**Issue by roberthunt** [sequelpro/sequelpro#2811]
I have noticed that dropping an index that includes a foreign key column automatically drops the foreign key as well, even if the index involves multiple columns, this behaviour is unexpected. Given a simple example:

```
CREATE TABLE `child` (
  `id` int(11) NOT NULL AUTO_INCREMENT,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB;

CREATE TABLE `parent` (
  `id` int(11) NOT NULL AUTO_INCREMENT,
  `child_id` int(11) NOT NULL,
  PRIMARY KEY (`id`),
  UNIQUE KEY `my_unique_key` (`id`,`child_id`),
  KEY `child_id` (`child_id`),
  CONSTRAINT `my_foreign_key` FOREIGN KEY (`child_id`) REFERENCES `child` (`id`)
) ENGINE=InnoDB;
```

Dropping the unique key on the parent table will drop the foreign key in **Sequel Pro 1.1.2 Build 4541 602e11a**

It depends which part of the unique key you select - selecting the non-foreign key column is okay:
![screen shot 2017-06-01 at 10 15 12](https://cloud.githubusercontent.com/assets/567676/26673048/6dab18f0-46b3-11e7-91f5-b5fe05edee0c.png)

Selecting the foreign key column seems to produce this inconsistent dialog (notice the discrepancy between the title and content) :
![screen shot 2017-06-01 at 10 15 32](https://cloud.githubusercontent.com/assets/567676/26673081/8b75ea22-46b3-11e7-9433-14902b2323a0.png)

**Comment by dmoagx** [comment]
Yeah, there is a quite silly bug in the formatting code that makes me wonder how that even compiles.
Anyway, the error message should read:

> The foreign key relationship 'my_foreign_key' has a dependency on this index. This relationship must be removed before the index can be deleted.
> 
> Are you sure you want to continue to delete the relationship and the index? This action cannot be undone.

However that error is wrong, too, since `my_unique_key` is a compound index and therefore couldn't even be used as the FK index (which is why MySQL added a second `child_id` index).

**Comment by braian87b** [comment]
Hi, When I add a foreing key on some column and unique constraint on the same column is only shown the unique constraint name on the list of indexes, and when try to delete it it shows the name of the foreing key at description, and If I add both on inverse order it happens the same, both are deleted, it should probably show both as two rows on the indexes list to avoid confussion, this is a bug related to this one... Am I right?

**Comment by dmoagx** [comment]
@braian87b Well, you can try a development build from https://sequelpro.com/test-builds to test this.
